maven-extension/src/main/java/io/opentelemetry/maven/OtelExecutionListener.java

```diff
@@ -45,6 +47,8 @@ public final class OtelExecutionListener extends AbstractExecutionListener {
 
   private static final Logger logger = LoggerFactory.getLogger(OtelExecutionListener.class);
 
+  private static final ThreadLocal<Scope> MOJO_EXECUTION_SCOPE = new ThreadLocal<>();
```

Comment:
Does this cause any issue with parallel Maven builds with `-T`?

Reply:
Yes, using a threadlocal from `ExecutionListener#mojoStarted()` to `ExecutionListener#mojoSucceeded()` / `ExecutionListener#mojoFailed()` works even when parallelising the Maven builds with `-T`.
The code contains state checks to ensure this assumption continues to be valid with newer versions of Maven parallelization.

Reply:
I just added a comment in the code to explain the rationale.